I11 new employee

Mefizz opened the pull request:
Фронт доповнений включно із вченим званням(далі що робота над кваліфікаціями)
Рішення із створенням спеціалізації з кваліфікацією буде фіналізовано після обговорення у понеділок з БА.

Ще потрібно буде валідувати на фронті і на беці.
Розбити логіку окремо на створення і редагування.
Таки зберегти всі данні констистентно в бд.
І досліджувати додаванням підпису.

У мене з @lemadior трошки різнилось локальне оточення, через що я не бачив редагування закладів.
Я вніс певні зміни, аби воно працювало у мене локально. Тому прошу звернути окрему увагу на 
app/Livewire/LegalEntity/EditLegalEntity.php:83
аби це комусь щось не зламало.

KarbovskyiV committed:
156 feat: Add information block, add validations,
refactor: move formatting moethod to separate service class,
remove unused phone methods from FormTrait class,
chore: add laravel-debugbar,
add docker related files and folders to gitignore

KarbovskyiV committed:
156 feat: Add information block, add validations,
refactor: move formatting moethod to separate service class,
remove unused phone methods from FormTrait class,
chore: add laravel-debugbar,
add docker related files and folders to gitignore

KarbovskyiV committed:
chore: composer update, add name, avoid unbound version constraints for laravel/fortify

Commit:
get updated dev

Commit:
you could access the property by using: "app(LegalEntityContext::class)""

KarbovskyiV committed:
create ServiceProvider that based on config decide which db models to use

Commit:
adding qualifications/education/document/general employee data. Working backend logic for saving data into DB.

Commit:
adding qualifications/education/document/general employee data. Working backend logic for saving data into DB.

Commit:
adding qualifications/education/document/general employee data. Working backend logic for saving data into DB.

KarbovskyiV committed:
attach evidences and vaccination protocol to relevant attributes,
remove static encounter call from seeder, out of necessity of one instance for format calls,
feat: add call test user seeder to only local env,
refactor: delete 1 unsed method, add few validate rule

dmkopylets committed:
to .gitignore added Makefile, docker-compose.yml files

Commit:
Merge branch 'dev' into i11_new_employee

Commit:
update for speciality & science_degree

dmkopylets committed:
rebased i11 to dev

dmkopylets committed:
synchronized with Mefizz

dmkopylets committed:
synchronized with Mefizz

dmkopylets committed:
synchronized with Mefizz

dmkopylets committed:
distinguish between Create and Edit and pass parameters

dmkopylets committed:
i11 crud

Mefizz committed:
Merge pull request #1 from dmkopylets/i11_new_employee_dk

I11_crud

Commit:
update for speciality & science_degree

Commit:
update for speciality & science_degree

Commit:
update for speciality & science_degree

Commit:
Merge remote-tracking branch 'origin/i11_new_employee' into i11_new_employee

# Conflicts:
#	composer.json
#	composer.lock

Commit:
add debugging tool

Commit:
cleaning composer.json

Commit:
#11 working on createEmployee logic

Commit:
#11 working on createEmployee logic

KarbovskyiV committed:
#135 refactor: split livewire component and view files to separate edit and create, and create basic component and view file

Commit:
#11 update for createEmployee form fields

Commit:
Merge branch 'dev' into i11_new_employee

# Conflicts:
#	.gitignore
#	app/Http/Kernel.php
#	app/Models/Employee/EmployeeRequest.php
#	app/Models/User.php
#	app/Providers/AppServiceProvider.php
#	composer.lock

Commit:
#11 form fields update

Commit:
#11 form fields update

Commit:
#11 form fields update

KarbovskyiV committed:
#135 refactor: add getFilteredKeysFromConfig method to FormTrait

KarbovskyiV committed:
#135 feat: add check and displaying error if same code was selected

KarbovskyiV committed:
#135 refactor: add initializeComponent method for init same data,
move public methods to the top of the component,
improve person db query,
small fixes,
chore: add rules to pint for imports

KarbovskyiV committed:
#170 feat: add coding system part for observations, remove borber line from modals

lemadior committed:
#i172 Fix issues after first login functinality has been merged

Commit:
#11 reworking save

dmkopylets committed:
added parameter passing  in app/Livewire/Employee/EmployeeEdit.php

dmkopylets committed:
added parameter passing  in app/Livewire/Employee/EmployeeEdit.php

dmkopylets committed:
fix preview commit

dmkopylets committed:
fixing dashboard/legal-entities/edit

lemadior committed:
#i172 Optimize role seeders to reduce DB requests

dmkopylets committed:
fixed:
resources/views/livewire/employee/_parts/_documents.blade.php
and
resources/views/livewire/employee/_parts/_qualifications.blade.php

dmkopylets committed:
fixed:
resources/views/livewire/employee/_parts/_documents.blade.php
and
resources/views/livewire/employee/_parts/_qualifications.blade.php

dmkopylets committed:
_specialities.blade.php fixed

dmkopylets committed:
_science_degree.blade.php WIP

Mefizz committed:
Merge pull request #6 from dmkopylets/i11_new_employee

I11 new employee modal window style fixing

lemadior committed:
#i172 Change callback functions to arrow ones

lemadior committed:
#i172 Modify Permission seeder to reduce DB requests as small as possible

Commit:
#11 update blade templates, extend models, add save EmployeeCreate.php logic

Commit:
#11 update blade templates, extend models, add save EmployeeCreate.php logic

Commit:
#11 update blade templates, extend models, add save EmployeeCreate.php logic

Commit:
#11 update blade templates, extend models, add save EmployeeCreate.php logic

Vitaliy-1 commented on app/Http/Middleware/SetLegalEntityContext.php:
```diff
@@ -0,0 +1,21 @@
+        $user = auth()->user();
+
+        if ($user && $user->legalEntity) {
+            app(LegalEntityContext::class)->set($user->legalEntity);
```
Тут треба ті зміни, які ми з вами вже обговорювали в попередньому PR

Vitaliy-1 commented on app/Livewire/Employee/EmployeeCreate.php:
```diff
@@ -2,22 +2,201 @@
+            $this->createPhones($party, $validated['party']['phones'] ?? []);
+
+            $this->createEmployeeRequest(
+                ['party_id' => $party->id,
```
Форматування:
```php
$this->createEmployeeRequest([
...
])
```

Vitaliy-1 commented on app/Livewire/Employee/EmployeeCreate.php:
```diff
@@ -2,22 +2,201 @@
+
+    protected function createUser(Person $person): User
+    {
+        DB::statement("SELECT setval('users_id_seq', (SELECT MAX(id) FROM users))");
```
Чи є можливість обійтись без raw queries? Або треба перевірити чи цей statement підтримується усіма SQL базами даних https://laravel.com/docs/12.x/database#introduction

Vitaliy-1 commented:
Тут в PR багато змін по front-end. Чи можете зробити rebase на dev, щоб була можливість протестувати?

Mefizz commented:
ребейзнув

Commit:
#11 finished saving employeeRequest backend

Commit:
#11 update front-end form

KarbovskyiV committed:
#170 feat: add main info block,
refactor: rename files to kebab case,
feat: add method do dictionary for getting child_values,
feat: add usage of InDictionary rule for few dictonary names,
chore: add style rules

KarbovskyiV committed:
refactor: rename files to kebab case,
feat: add method do dictionary for getting child_values,
feat: add usage of InDictionary rule for few dictonary names,
chore: add style rules

lemadior committed:
i#i167 Edit Legal Entity

lemadior committed:
i#i167 Edit Legal Entity

lemadior committed:
i#i167 Edit Legal Entity

KarbovskyiV committed:
#170 feat: add additional info block

KarbovskyiV committed:
#170 fix: change data type of reasons not given and evidences to array,
feat: add comment block and set default datetime in observation,
fix: get keys in InDictionary

lemadior committed:
#i167 Fix impossibility to log in after creating the LegalEntity

Commit:
#11 refactor LegalEntityContext.php

Commit:
#11 refactor LegalEntityContext.php

Commit:
#11 refactor LegalEntityContext.php

Commit:
#11 refactor LegalEntityContext.php

Commit:
#11 refactor save employee request

Mefizz commented:
@Vitaliy-1 долив зміни, ребейзнувсь, порізолвив купу всього.

Commit:
#11 revert Encounter component naming

Commit:
Merge branch 'dev' into i11_new_employee

# Conflicts:
#	.gitignore
#	app/Livewire/Encounter/EncounterComponent.php
#	app/Models/Employee/BaseEmployee.php
#	app/Models/Employee/EmployeeRequest.php
#	app/Models/Relations/Speciality.php
#	app/Repositories/ScienceDegreeRepository.php

Commit:
#11 fix after rebase and conflicts resolving

lemadior commented on app/Livewire/Employee/EmployeeForm.php:
```diff
@@ -5,18 +5,22 @@
 use App\Classes\eHealth\Api\EmployeeApi;
 use App\Livewire\Employee\Forms\Api\EmployeeRequestApi;
 use App\Livewire\Employee\Forms\EmployeeForm as Form;
+use App\Livewire\LegalEntity\LegalEntity;
```
Тут потрібен псевдонім. Нижче вже є імпорт класу use App\Models\LegalEntity;

lemadior commented on app/Models/Employee/EmployeeRequest.php:
```diff
@@ -21,4 +25,13 @@ public function employee(): BelongsTo
+    {
+        static::creating(function ($employeeRequest) {
+            if (empty($employeeRequest->uuid)) {
+                $employeeRequest->uuid = (string)Str::uuid();
```
Навіщо це? $employeeRequest->uuid завжди має заповнюватись з даних, отриманих від сервера!

Mefizz replied:
а на етапі збереження?
чернетки запиту до нас в бд?

lemadior replied:
Так

lemadior commented on app/Models/Relations/Party.php:
```diff
@@ -28,11 +31,20 @@ class Party extends Model
+    {
+        static::creating(function ($party) {
+            if (empty($party->uuid)) {
+                $party->uuid = (string) Str::uuid();
```
Party не має містити довільний uuid. Якщо його немає - там має бути null. Потім, при заході користувача в систему - він перепишеться валідним значенням, отриманим від сервера

lemadior reviewed (commented):
Потрібно окремо обговорити ще ряд важливих моментів!

Vitaliy-1 commented on .gitignore:
```diff
@@ -18,6 +18,10 @@ yarn-error.log
 _ide_helper_models.php
-storage/debugbar
+/storage/debugbar
+composer.lock
```
`composer.lock` додавати в gitignore не треба. По ньому підтягуються конкретні перевірені версії залежностей.

Vitaliy-1 commented on .env.example:
```diff
@@ -71,3 +71,8 @@ CIPHER_API_URL=
+WWWUSER=1000
+WWWGROUP=1000
+
+SAIL_XDEBUG_MODE=develop,debug,coverage
```
Конфігурації Laravel Sail додавати в dev не треба

Vitaliy-1 commented on app/Http/Kernel.php:
```diff
@@ -36,8 +36,9 @@ class Kernel extends HttpKernel
+            \App\Http\Middleware\SetLegalEntityContext::class,
             \App\Classes\eHealth\Middleware\CheckSessionToken::class,
-
+            \App\Http\Middleware\SetLegalEntityContext::class,
```
Поки приберіть все що повʼязано із привʼязкою Legal Entity до Laravel Container із цього PR

Vitaliy-1 commented on app/Http/Middleware/SetLegalEntityContext.php:
Це краще розглянути в рамках окремого PR

Vitaliy-1 commented on app/Livewire/Employee/EmployeeComponent.php:
```diff
@@ -25,7 +26,12 @@ class EmployeeComponent extends Component
+    /**
+     * @var LegalEntityContext
+     */
+    protected LegalEntityContext $legalEntityContext;
```
Див вище

Vitaliy-1 commented on app/Livewire/Employee/EmployeeComponent.php:
```diff
@@ -54,15 +60,16 @@ class EmployeeComponent extends Component
     public array $employeeTypePosition = [];
 
-    public function mount(): void
+    public function boot(EmployeeRepository $employeeRepository, LegalEntityContext $legalEntityContext): void
```
Див вище + після рефакторингу Legal Entity вже буде привʼязаний до контейнера і необхідності додатково привʼязувати його до компонента не буде.

Vitaliy-1 commented on app/Livewire/Employee/EmployeeCreate.php:
В цьому PR хай цей патерн залишається, але в майбутньому слід використовувати вже написаний Repository для employee

Vitaliy-1 commented on tests/CreatesApplication.php:
В рамках цього PR зміни в дефолтних тестах робити не треба. Це ми вирішимо окремо

Vitaliy-1 commented on app/Livewire/Employee/Forms/EmployeeForm.php:
```diff
@@ -2,146 +2,191 @@
-        'employeeType' => '',
-        'phones' => [
+    public array $party = [
+        'position'      => '',
```
Ми подібне форматування не використовуємо

Vitaliy-1 commented on app/Livewire/Employee/Forms/EmployeeForm.php:
```diff
@@ -2,146 +2,191 @@
 
     /**
-     * @throws ValidationException
+     * Валідація конкретного блоку форми (не повна форма)
```
В коді коментарі робимо англійською. В окремій документації будемо писати українською.

Vitaliy-1 commented on app/Services/EncounterService.php:
Артефакт rebase - слід виправити

Vitaliy-1 reviewed (changes requested):
Дякую! Залишив пару коментарів. там основне - є помилки при rebase + деякі зміни слід робити в окремих PR, оскільки потребують більш детального обговорення та додаткових змін.
